src/commands/deploy.rs

```diff
@@ -238,15 +238,121 @@ impl DeployCommand {
+
+        let cfg_any = spin_loader::local::raw_manifest_from_file(&self.app).await?;
+        let RawAppManifestAnyVersion::V1(cfg) = cfg_any;
+
+        let buildinfo = if !self.no_buildinfo {
+            match &self.buildinfo {
+                Some(i) => Some(i.clone()),
+                None => self.compute_buildinfo(&cfg).await.map(Option::Some)?,
+            }
+        } else {
+            None
+        };
+
+        let bindle_connection_info = BindleConnectionInfo::from_token(
+            format!("{}/{}", login_connection.url, BINDLE_REGISTRY_URL_PATH),
+            login_connection.danger_accept_invalid_certs,
+            login_connection.token,
+        );
+
+        let bindle_id = self
+            .create_and_push_bindle(buildinfo, bindle_connection_info)
+            .await?;
+        let name = bindle_id.name().to_string();
+
+        // Create or update app
+        // TODO: this process involves many calls to Hippo. Should be able to update the channel
+        // via only `add_revision` if bindle naming schema is updated so bindles can be deterministically ordered by Hippo.
+        let channel_id = match self.get_app_id_cloud(&client, name.clone()).await {
+            Ok(app_id) => {
+                CloudClient::add_revision(
+                    &client,
+                    name.clone(),
+                    bindle_id.version_string().clone(),
+                )
+                .await?;
+                let existing_channel_id = self
+                    .get_channel_id_cloud(&client, SPIN_DEPLOY_CHANNEL_NAME.to_string(), app_id)
+                    .await?;
+                let active_revision_id = self
+                    .get_revision_id_cloud(&client, bindle_id.version_string().clone(), app_id)
+                    .await?;
+                CloudClient::patch_channel(
+                    &client,
+                    existing_channel_id,
+                    None,
+                    Some(CloudChannelRevisionSelectionStrategy::UseSpecifiedRevision),
+                    None,
+                    Some(active_revision_id),
+                    None,
+                )
+                .await
+                .context("Problem patching a channel")?;
+
+                existing_channel_id
+            }
+            Err(_) => {
+                let range_rule = Some(bindle_id.version_string());
+                let app_id = CloudClient::add_app(&client, &name, &name)
+                    .await
+                    .context("Unable to create app")?;
+                CloudClient::add_channel(
+                    &client,
+                    app_id,
+                    String::from(SPIN_DEPLOY_CHANNEL_NAME),
+                    CloudChannelRevisionSelectionStrategy::UseRangeRule,
+                    range_rule,
+                    None,
+                )
+                .await
+                .context("Problem creating a channel")?
+            }
+        };
+
+        println!(
```

Comment:
The last rebase introduced a call to `wait_for_ready` on the Hippo flow - flagging this up for the Cloud flow too.  (Sorry, I should have looped you in on the PR where that was introduced.)